pkg/prowgen/prowgen_tests.go

```diff
@@ -181,14 +183,31 @@ func createTest(r Repository, line string, shouldMatch string, tests *[]Test, on
 	return nil
 }
 
-func dependenciesFromImages(images []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration) []cioperatorapi.StepDependency {
+func dependenciesFromImages(r Repository, images []cioperatorapi.ProjectDirectoryImageBuildStepConfiguration, testName string) []cioperatorapi.StepDependency {
 	deps := make([]cioperatorapi.StepDependency, 0, len(images))
 	for _, image := range images {
-		dep := cioperatorapi.StepDependency{
-			Name: strings.ReplaceAll(string(image.To), "_", "-"),
-			Env:  strings.ToUpper(strings.ReplaceAll(string(image.To), "-", "_")),
+		imageFinal := strings.ReplaceAll(string(image.To), "_", "-")
+		if shouldAcceptImage(testName, r.E2ETests.FilterImages, imageFinal) {
+			dep := cioperatorapi.StepDependency{
+				Name: imageFinal,
+				Env:  strings.ToUpper(strings.ReplaceAll(string(image.To), "-", "_")),
+			}
+			deps = append(deps, dep)
 		}
-		deps = append(deps, dep)
 	}
 	return deps
 }
+
+// If an image is found to be part of a filtering rule and the test name does not satisfy the prefix condition
+// the image is rejected. In any other case it is accepted.
```

Comment:
In the future this can be relaxed to have an or logic but now it is good enough.
The idea is to have images related 1-1 to a subset/class of tests only eg. perf images, security images etc.